benchmarks/PHeapSort.hs

```diff
@@ -0,0 +1,11 @@
+module PHeapSort where
+import Data.PQueue.Prio.Min (MinPQueue)
+import qualified Data.PQueue.Prio.Min as P
+import System.Random
+import Data.Int
+
+heapSortRandoms :: Int -> StdGen -> [Int32]
+heapSortRandoms n gen = heapSort $ take n (randoms gen)
+
+heapSort :: Ord a => [a] -> [a]
+heapSort xs = [b | (b, ~()) <- P.toAscList . P.fromList . map (\a -> (a, ())) $ xs]
```

Comment:
I guess it shouldn't matter, because an outer `map` should fuse with `toAscList` (I *think* the latter has a rewrite rule) and avoid producing thunks to select the first components. But if that doesn't happen, things won't look great because `rnf` for lists doesn't fuse (maybe the assumption is that if you `rnf` something then you'll use it again, but that's not a great assumption).